todoman/ui.py

```diff
@@ -258,41 +283,39 @@ def detailed(self, todo):
             rv = "{}\n\n{}".format(rv, todo.description)
         return rv
 
-    def format_date(self, date):
-        """
-        Returns date in the following format:
-        * if date == today or tomorrow: "Today" or "Tomorrow"
-        * else: return a string representing that date
-        * if no date is supplied, it returns empty_date
-
-        :param datetime.datetime date: a datetime object
-        """
-        if date:
-            assert isinstance(date, datetime)
-            if date.date() in self.special_dates:
-                rv = self.special_dates[date.date()]
-            else:
-                rv = date.strftime(self.date_format)
-            return rv
+    def format_datetime(self, dt):
+        dt = dt or None
+        if dt in self.special_datetimes:
+            format_string = self.special_datetimes[dt]
+        elif dt.date() in self.special_datetimes:
+            format_string = self.special_datetimes[dt.date()]
         else:
-            return self.empty_date
+            format_string = self.datetime_format
+        if not dt:
+            return format_string
+        return dt.strftime(format_string)
 
-    def parse_date(self, date):
-        if not date:
+    def parse_datetime(self, dt):
+        if not dt:
             return None
 
-        try:
-            rv = datetime.strptime(date, self.date_format)
-        except ValueError:
-            rv, certainty = self._parsedatetime_calendar.parse(date)
-            if not certainty:
-                raise ValueError(
-                    'Time description not recognized: {}' .format(date)
-                )
-            rv = datetime.fromtimestamp(mktime(rv))
-
+        rv = self._parse_datetime_naive(dt)
```

Comment:
`_parse_datetime_naive` has multiple return paths, those would all have to add the tzinfo. That's the only reason for the split. (BTW the parsing functions didn't change in my recent commits, they've been like that from the beginning)